### PR TITLE
swish-test updates and tilde-expansion for filesystem operations

### PR DESCRIPTION
**Pull request by owaddell**
**Proposed changes**

Add `swish-test` options to make it easier for folks to generate a text-based summary of prior test results or to re-run prior tests that failed or were skipped. Handle limited (current-user) form of tilde expansion for filesystem operations.

* `swish-test`: allow `--progress` in combination with `--harvest` or `--load-results`.
* `swish-test`: add `--rerun` option
* `swish-test`: include exit reason when reporting failure to load profile data
* add `osi_get_home_directory`
* support `~` and paths with `~/` prefix in filesystem procedures

### CODE REVIEW

File: src/swish/swish-test.ms [hunk 1]
```diff
@@ -228,8 +240,9 @@
   )
 
 (isolate-mat select-tests ()
+  (define subdir (uuid->string (osi_make_uuid)))
   (define test-file
-    (write-test-file "test-specified.ms"
+    (write-test-file (path-combine "test-specified.ms")
```
Comment (burgerrg) [inline src/swish/swish-test.ms]: Are you missing the path argument to path-combine?

Comment (owaddell) [reply src/swish/swish-test.ms]: Well spotted. I meant to use the subdir to check the case where we `--rerun` all results in a particular directory.

Pushed a fix for that along with some proposed breaking changes.

**Review by burgerrg — APPROVED** [review]
Thanks for these useful changes!

**Review by laqrix — APPROVED** [review]
Looks good. Thanks Oscar!

**Comment by owaddell** [comment]
Thanks for the reviews, folks!

**Pull request merged**